File: perl/parse_yaml.pl
#!/usr/local/bin/perl

use strict;
use YAML::Tiny;
use Data::Dumper;

my $yaml = '#cloud-config
hostname: foo
password: hoge
ssh_pwauth: True
chpasswd:
    expire: False
';

my $data = YAML::Tiny->new;
$data = YAML::Tiny->read_string($yaml);
print $yaml;
print Dumper($data);

print $data->[0]->{password};

File: php/parse_yaml.php
<?php

$yaml = '#cloud-config
hostname: foo
password: hoge
ssh_pwauth: True
chpasswd:
    expire: False
';

$yaml_abnormal = '#cloud-config
hostname: foopassword: hoge
ssh_pwauth: True
chpasswd:
    expire: False
';

$parsed_yaml = yaml_parse($yaml);

// var_dump($yaml);
var_dump($parsed_yaml);

if ($parsed_yaml == false) {
    var_dump("parse error");
}
